smoke/tests/tool/nydusd.go

```diff
@@ -315,19 +315,24 @@ func (nydusd *Nydusd) Mount() error {
 }
 
 func (nydusd *Nydusd) MountByAPI(config NydusdConfig) error {
-	err := makeConfig(NydusdConfigTpl, config)
-	if err != nil {
-		return err
-	}
-	f, err := os.Open(config.ConfigPath)
-	if err != nil {
-		return err
+	var tpl *template.Template
```

Comment:
Why we modify this?

Reply:
This is to remove the makeConfig function so that the API can be used directly instead of first writing to a file and then using the API.